src/test/build-all-projects.ts

```diff
@@ -27,49 +29,55 @@ describe('Setup test environment for Yo Office build tests', () => {
 // Build Typescript project types for all supported hosts
 for (var i = 0; i < hostsTemplates.length; i++)
 {
-    for (var j = 0; j < tsTemplates.length; j++)
+    for (var j = 0; j < projectTemplates.length; j++)
```

Comment:
>for (var j = 0; j < projectTemplates.length; j++) [](start = 4, length = 49)

I think that the build Typescript and Javascript logic can be refactored and execute it within this loop (instead of creating two more nested loops), the only difference seems to be the use of the **javascript** and **typescript** const, maybe you can make a function which receives the args and then make the rest of the logic using it, e.g. 

if (parsedProjectJsonData.projectTypes[projectTemplates[j]].typescript){
    build(projectTemplates[j], hostTemplates[i], typescript);
}
if (parsedProjectJsonData.projectTypes[projectTemplates[j]].javascript){
    build(projectTemplates[j], hostTemplates[i], javascript);
}